sdk/resourcemanager/Azure.ResourceManager/tests/Scenario/LroRehydrationTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Tests
+{
+    public class LroRehydrationTests : ResourceManagerTestBase
+    {
+        public LroRehydrationTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task CreateDeleteResourceGroup()
+        {
+            string rgName = Recording.GenerateAssetName("testRg-");
+            var tags = new Dictionary<string, string>()
+            {
+                { "key", "value"}
+            };
+            SubscriptionResource subscription = await Client.GetDefaultSubscriptionAsync().ConfigureAwait(false);
+            // The creation of a resource group is a fake LRO
+            var rgOp = await subscription.GetResourceGroups().Construct(AzureLocation.WestUS2, tags).CreateOrUpdateAsync(rgName);
+            var rgOpId = rgOp.Id;
+            var rehydration = new ResourcesArmOperationRehydration<ResourceGroupResource>(Client, rgOpId);
+            var rehydratedLro = await rehydration.RehydrateAsync(WaitUntil.Completed);
+            Assert.AreEqual(rehydratedLro.Id, rgOpId);
+            ResourceGroupResource rg = rehydratedLro.Value;
+            Assert.AreEqual(rgName, rg.Data.Name);
```

Comment:
can we add full body comparison, potentially in raw json format if thats easiest